### ISSUE TITLE
Learning Rate Finder stops early because max epochs reached

### ISSUE THREAD

**Issue by DominikSpiljak** [Lightning-AI/lightning#16561]
### Bug description

If `max_epoch` is set to less than number of steps in `lr_find` function, the `lr_finder` is stopped with message: 
``` `Trainer.fit` stopped: `max_epochs=x` reached.```

### How to reproduce the bug

```python
python
import os

import torch
from pytorch_lightning import LightningModule, Trainer, callbacks
from torch.utils.data import DataLoader, Dataset


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("train_loss", loss)
        return {"loss": loss}

    def validation_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("valid_loss", loss)

    def test_step(self, batch, batch_idx):
        loss = self(batch).sum()
        self.log("test_loss", loss)

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


class EpochAutoLRFinder(callbacks.LearningRateFinder):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def on_fit_start(self, trainer, pl_module) -> None:
        return

    def on_train_epoch_start(self, trainer, pl_module) -> None:
        self.lr_find(trainer, pl_module)

        if (
            self.optimal_lr is not None
            and (lr_suggestion := self.optimal_lr.suggestion()) is not None
        ):
            for group in trainer.optimizers[0].param_groups:
                group["lr"] = lr_suggestion


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    val_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    test_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        limit_train_batches=100,
        limit_val_batches=100,
        limit_test_batches=100,
        num_sanity_val_steps=0,
        max_epochs=20,
        enable_model_summary=False,
        callbacks=[EpochAutoLRFinder()],
    )
    trainer.fit(model, train_dataloaders=train_data, val_dataloaders=val_data)
    trainer.test(model, dataloaders=test_data)


if __name__ == "__main__":
    run()
```

cc @borda

**Comment by awaelchli** [comment]
This could potentially be solved by a check in the LR finder before running the trainer.
